pkg/collector/corechecks/snmp/internal/devicecheck/devicecheck_test.go

```diff
@@ -561,11 +563,21 @@ profiles:
 	deviceCk, err := NewDeviceCheck(config, "1.2.3.4", sessionFactory)
 	assert.Nil(t, err)
 
+	snmpTags := []string{"snmp_device:1.2.3.4", "snmp_profile:f5-big-ip", "device_vendor:f5", "snmp_host:foo_sys_name",
+		"static_tag:from_profile_root", "static_tag:from_profile_root", "some_tag:some_tag_value", "prefix:f", "suffix:oo_sys_name"}
+
 	sender := mocksender.NewMockSender("123") // required to initiate aggregator
 	sender.On("Gauge", mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return()
 	sender.On("MonotonicCount", mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return()
 	sender.On("ServiceCheck", mock.Anything, mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return()
-	sender.On("EventPlatformEvent", mock.Anything, mock.Anything).Return()
+	sender.On("EventPlatformEvent", mock.Anything, mock.Anything).Return().Run(func(args mock.Arguments) {
+		arg := args.Get(0).([]uint8)
+		var data metadata.NetworkDevicesMetadata
+		json.Unmarshal(arg, &data)
+
+		tags := data.Devices[0].Tags
+		assert.Subset(t, tags, snmpTags)
```

Comment:
Can we assert the whole array instead of asserting that it's a subset?

Reply:
It is not possible because a tag with agent_version is added in the metadata tags which is not present in the metric tags